lib/sha256.c

```diff
@@ -105,8 +106,8 @@ typedef mbedtls_sha256_context SHA256_CTX;
 
 static void SHA256_Init(SHA256_CTX *ctx)
 {
-#if !defined(HAS_RESULT_CODE_BASED_FUNCTIONS)
-  mbedtls_sha256_starts(ctx, 0);
+#if !defined(HAS_MBEDTLS_RESULT_CODE_BASED_FUNCTIONS)
```

Comment:
Is there a style guide for `#if !defined...` vs `#ifndef`?

https://github.com/curl/curl/search?q=ifndef it looks like both flavors are used...

offhand, it looks like `#if !defined...` tends to be used with a boolean operator for when one is performing multiple checks concurrently.
https://github.com/curl/curl/search?p=2&q=%22%23if+%21defined%22

(Yes, I understand this is preexisting code.)

Reply:
> Is there a style guide for #if !defined... vs #ifndef?

We prefer #ifdef and #ifndef if you only check a single define. If you check more than one, then #if needs to be used.